### DISCUSSION TITLE
Unable to import external React libraries whose components import scss files

### DISCUSSION THREAD

**Discussion by bestuiexperience** [vercel/next.js#13781] [Help]
I am trying to import a package into my Next JS project whose components import scss files. However, I get the following error:

document is not defined
ReferenceError: document is not defined
    at i (C:\node_modules path to imported libary\dist\main.js:)

It is very important to be able to import eternal packages that import scss files this way. Is there any workaround or fix for this issue?

**Comment by rafaelalmeidatk** [comment]
The issue seems to be with the package code that is trying to access `document`. Next will render your app in the Node.js environment where `document` isn't defined so you can't use it anywhere. If you share the code of this package where `document` is utilized we may help fixing it.

**Comment by bestuiexperience** [comment]
I can't share it because the library is propriety. But here is a snippet that will give you an idea. Suppose the package I am trying to import is called ' best-ui-library' (a fictitious name) and suppose it has a react component, Button.jsx)  that begins as follows:

import React from 'react';
import PropTypes from 'prop-types';
import './Button.scss';

------- more code here

export default Button;

-------------------------------------------

Now in my Next JS react component I import the button as follows:

import Button from ' best-ui-library';

When I start the server I get the error mentioned above.

If the code in Button.jsx is modified to eliminate import './Button.scss', then the error goes away.

**Reply by jmvelasquezatwongdoody** [reply]
any solutions?
